src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1615,7 +1615,12 @@ internal StackValue DispatchDispose(StackValue stackValue, RuntimeCore runtimeCo
             //Recursion base case
             if (replicationInstructions.Count == 0)
             {
-                return ExecWithZeroRI(functionEndPoint, c, formalParameters, stackFrame, runtimeCore, previousTraceData, newTraceData, finalFunctionEndPoint);
+                return ExecWithZeroRI(functionEndPoints, c, formalParameters, stackFrame, runtimeCore, previousTraceData, newTraceData, finalFunctionEndPoint);
+            }
+
+            if (finalFunctionEndPoint == null && functionEndPoints.Count == 1)
+            {
+                finalFunctionEndPoint = SelectFinalFep(c, functionEndPoints, formalParameters, stackFrame, runtimeCore);
```

Comment:
What's the logic of determining the final function endpoint on the basis of `functionEndPoints` count vs. homogeneous parameters? I don't quite understand how one check can be swapped for another. The two seem to be different things, don't they?

Reply:
Ah, does this mean that previously you determined how many overloads of a function needed to be processed depending on the types of parameters whereas now you just check for the number of overloads?

Reply:
If there's just one `functionEndPoint`, can't we just say `finalFunctionEndPoint = functionEndPoints[0]`?

Reply:
Never mind, I guess it could be `null`.

Reply:
@aparajit-pratap Yes you second comment is correct.  `AreParametersHomogeneousTypes` didn't check how may overloads functions were required.  It just check if all the types were the same and then inferred no overloads were required.   What I realized in testing is that the functionsEndPoints.Count was already determining the same information and a better test of the same result.  The `AreParametersHomogeneousTypes` check is now superfluous.  If you only have one option return from `ComputeFeps()` which means `functionsEndpoints.Count == 1` then it is safe to assign the functionEndPoint once for the life of the `Callsite` replication passes.